scripts/template_tester.py

```diff
@@ -17,9 +17,19 @@
 from utils import trackerbot
 
 
-def get(api):
+def get(api, request_type=None):
     try:
-        template, provider_key, stream = trackerbot.templates_to_test(api, limit=1)[0]
+        template, provider_key, stream, provider_type =\
+            trackerbot.templates_to_test(api, limit=1)[0]
+        if request_type == 'rhevm' and provider_type != 'rhevm':
+            # No untested templates of this provider-type
+            return 0
+        elif request_type == 'openstack' and provider_type != 'openstack':
+            # No untested templates of this provider-type
+            return 0
+        elif request_type == 'vsphere' and provider_type != 'vsphere':
+            # No untested templates of this provider-type
+            return 0
```

Comment:
This won't achieve parallelization unfortunately. Take the case of a vsphere provider being returned as the single item. In this case, the vsphere job will start work, but the rhevm and openstack jobs will block because the vsphere won't be cleared until the template tester completes.

What we need to do is somehow bake in the request type to the templates_to_test api call, so that we can request a vsphere, or a rhevm or an openstack. In this way the blocking will not occur.